### PR TITLE
Refs [TASK][35850] Create ui product type screen

### PR DESCRIPTION
**Pull request by chiennn19122000**
## Related Tickets
- [#Task 35850: Create ui product type screen](https://edu-redmine.sun-asterisk.vn/issues/35850)
## WHAT
- 
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/makeupwindow/data/model/Product.kt [hunk 1]
```diff
@@ -26,8 +26,12 @@ data class Product(
         jsonObject.getInt(ID),
         jsonObject.getString(BRAND),
         jsonObject.getString(NAME),
-        jsonObject.getString(PRICE),
-        if (jsonObject.getString(PRICESIGN) == NULL) PRICESIGN_DEFAULT else jsonObject.getString(
+        if (jsonObject.getString(PRICE)
+                .equals(NULL)
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/makeupwindow/data/model/Product.kt]: dùng == đi

File: app/src/main/java/com/sun/makeupwindow/ui/home/HomeFragment.kt [hunk 2]
```diff
@@ -125,6 +126,10 @@ class HomeFragment : BaseFragment(), HomeContract.View {
     }
 
     private fun itemCategoryClicked(category: Category) {
+        fragmentManager?.addFragment(
+            R.id.frameContainer,
+            ProductTypeFragment.getInstance(category)
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/makeupwindow/ui/home/HomeFragment.kt]: sao fragment lại getInstance?

File: app/src/main/java/com/sun/makeupwindow/ui/producttype/DialogPrice.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    private fun initAction(view: View) {
+        view.buttonCancel.setOnClickListener {
+            dismiss()
+        }
+
+        view.buttonApply.setOnClickListener {
+            val selectId = radioGroup.checkedRadioButtonId
+            val radio = view.findViewById<RadioButton>(selectId)
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/makeupwindow/ui/producttype/DialogPrice.kt]: dùng viewbinding đi

File: app/src/main/java/com/sun/makeupwindow/ui/producttype/ProductTypeFragment.kt [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+import com.sun.makeupwindow.ui.productdetail.ProductDetailFragment
+import com.sun.makeupwindow.utlis.*
+import kotlinx.android.synthetic.main.fragment_detail_product.*
+import kotlinx.android.synthetic.main.fragment_product_type.*
+import kotlinx.android.synthetic.main.fragment_product_type.buttonBackDetail
+
+class ProductTypeFragment : BaseFragment(), ProductTypeContract.View {
+
+    private var presenter: ProductTypePresenter? = null
+    private var adapter = ProductTypeAdapter(this::itemProductClicked)
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/makeupwindow/ui/producttype/ProductTypeFragment.kt]: val

File: app/src/main/java/com/sun/makeupwindow/ui/producttype/ProductTypeFragment.kt [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    private var loadingDialog: LoadingDialog? = null
+    private var category: Category? = null
+    private var dialogPrice = DialogPrice()
+    private var products = mutableListOf<Product>()
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/makeupwindow/ui/producttype/ProductTypeFragment.kt]: val

File: app/src/main/java/com/sun/makeupwindow/ui/producttype/ProductTypeFragment.kt [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+            }
+        }
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        if (requestCode == REQUEST_CODE) {
+            if (resultCode == RESULT_CODE) {
+                val price = data?.getStringExtra(RETURN_PRICE).toString()
+                when {
+                    price.equals(getString(R.string.text_all)) -> {
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/makeupwindow/ui/producttype/ProductTypeFragment.kt]: `==`

File: app/src/main/res/layout/dialog_price.xml [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+            android:text="@string/text_all"
+            android:checked="true"
+            android:textSize="@dimen/dp_20" />
+
+        <RadioButton
+            android:id="@+id/radioPrice1"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="@string/text_price_1"
+            android:textSize="@dimen/dp_20" />
```
Comment (datdt-1935) [inline app/src/main/res/layout/dialog_price.xml]: textSize sao lại dp?

Comment (chiennn19122000) [reply app/src/main/res/layout/dialog_price.xml]: cái này do em nhầm rồi ạ

File: app/src/main/java/com/sun/makeupwindow/ui/home/HomeFragment.kt (hunk 2 above)
Comment (chiennn19122000) [reply app/src/main/java/com/sun/makeupwindow/ui/home/HomeFragment.kt]: anh có thể cho em được tại sao không dùng getInstance đối với fragment không ạ

Comment (chiennn19122000) [reply app/src/main/java/com/sun/makeupwindow/ui/home/HomeFragment.kt]: à, bên trong companion object của em là 
companion object {
        private const val BUNDLE_PRODUCT_DETAIL = "BUNDLE_PRODUCT_DETAIL"

        fun getInstance(product: Product) = ProductDetailFragment().apply {
            arguments = bundleOf(BUNDLE_PRODUCT_DETAIL to product)
        }
    }

Comment (chiennn19122000) [reply app/src/main/java/com/sun/makeupwindow/ui/home/HomeFragment.kt]: em sẽ đổi tên lại là newInstance

File: app/src/main/java/com/sun/makeupwindow/ui/producttype/DialogPrice.kt [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+import com.sun.makeupwindow.utlis.RETURN_PRICE
+
+class DialogPrice : DialogFragment() {
+    private lateinit var binding: DialogPriceBinding
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding= DialogPriceBinding.inflate(inflater,container, false)
```
Comment (thonc-2135) [inline app/src/main/java/com/sun/makeupwindow/ui/producttype/DialogPrice.kt:22]: format

File: app/src/main/java/com/sun/makeupwindow/ui/producttype/ProductTypeFragment.kt [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+
+    override fun initData() {
+        initInputData()
+        initAdapter()
+        initPresenter()
+        initDialog()
+    }
+
+    private fun initAdapter() {
+        recyclerProductType.adapter = adapter
```
Comment (thonc-2135) [inline app/src/main/java/com/sun/makeupwindow/ui/producttype/ProductTypeFragment.kt]: dùng viewbinding

**Pull request merged**